app/controllers/requested_accounts_controller.rb

```diff
@@ -62,15 +73,32 @@ def create_accounts
     end
   end
 
+  def create_all_accounts
+    raise_unauthorized_exception unless user_signed_in? && current_user.admin?
+    requested_accounts = @courses.map(&:requested_accounts).flatten
+    @results = requested_accounts.inject([]) do |results, account|
+      results << create_account(account)
+    end
+
+    render :index
+  end
+
   private
 
+  def set_courses_with_requested_accounts
+    has_requested_accounts = ->(course) { course.requested_accounts.present? }
+    @courses = Course.current_and_future.select(&has_requested_accounts)
```

Comment:
An alternative approach could be to go through `RequestedAccount` and group by course. Then you only need to load the course records for the ones that have requested accounts, which are going to be a very small fraction of all courses.